### PR TITLE
feat: new package icons

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Feature

#### What is expected?

✨ New package `@ultraviolet/icons` that will include `Icon`, `ProductIcon` and `CategoryIcon` components.

> **Warning**
> `Icon` component from `@ultraviolet/ui` is now deprecated and we should use the same component from `@ultraviolet/icons` instead.

## Relevant logs and/or screenshots

Screenshot 2023-07-13 at 14 40 22
Screenshot 2023-07-13 at 14 40 29
Screenshot 2023-07-13 at 14 40 35

---

`@ultraviolet/ui` Icon component marked as deprecated:
Screenshot 2023-07-13 at 14 42 33

### CODE REVIEW

File: .changeset/light-moose-double.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@ultraviolet/ui': minor
```
Comment (chambo-e) [inline .changeset/light-moose-double.md:2]: It's a breaking theoretically 😐

File: packages/icons/package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  },
+  "os": [
+    "darwin",
+    "linux"
+  ],
+  "sideEffects": false,
+  "main": "dist/src/index.js",
+  "module": "dist/src/index.js",
+  "jsnext:main": "dist/src/index.js",
+  "types": "dist/index.d.ts",
```
Comment (chambo-e) [inline packages/icons/package.json:32]: Can we use directly the new export format from @QuiiBz PR ?

File: packages/icons/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  "peerDependencies": {
+    "@emotion/react": "11.11.1",
+    "@emotion/styled": "11.11.0",
+    "react": "18.2.0",
+    "react-dom": "18.2.0"
```
Comment (chambo-e) [inline packages/icons/package.json:37]: unused no ?

File: packages/icons/package.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  },
+  "devDependencies": {
+    "@babel/core": "7.22.8",
+    "@emotion/babel-plugin": "11.11.0",
+    "@emotion/react": "11.11.1",
+    "@emotion/styled": "11.11.0",
+    "@types/react": "18.2.8",
+    "@types/react-dom": "18.2.4",
+    "react": "18.2.0",
```
Comment (chambo-e) [inline packages/icons/package.json:46]: @emotion and react are already in dependencies

File: packages/icons/src/components/ProductIcon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+})<{ variant: Variants; disabled?: boolean; size: keyof typeof SIZES }>`
+  & {
+    width: ${({ size }) => `${SIZES[size]}px`};
+    height: ${({ size }) => `${SIZES[size]}px`};
+    min-width: ${({ size }) => `${SIZES[size]}px`};
+    max-height: ${({ size }) => `${SIZES[size]}px`};
+  }
+
+  .fill,
+  .fill > * {
```
Comment (chambo-e) [inline packages/icons/src/components/ProductIcon/index.tsx:31]: Are we sure abut the wildcard selector ?

File: packages/icons/src/components/ProductIcon/index.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const StyledIcon = styled('svg', {
+  shouldForwardProp: prop => !['variant', 'disabled'].includes(prop),
+})<{ variant: Variants; disabled?: boolean; size: keyof typeof SIZES }>`
+  & {
+    width: ${({ size }) => `${SIZES[size]}px`};
+    height: ${({ size }) => `${SIZES[size]}px`};
+    min-width: ${({ size }) => `${SIZES[size]}px`};
+    max-height: ${({ size }) => `${SIZES[size]}px`};
```
Comment (chambo-e) [inline packages/icons/src/components/ProductIcon/index.tsx:27]: min and max are redundant no ? width and height will already set the size

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: .changeset/light-moose-double.md (hunk 1 above)
Comment (matthprost) [reply .changeset/light-moose-double.md:2]: @chambo-e why is it breaking ? I didn't change anything on this package except to make Icon deprecated

File: packages/icons/package.json (hunk 2 above)
Comment (philibea) [reply packages/icons/package.json:32]: https://github.com/scaleway/scaleway-lib/pull/1380

File: packages/icons/src/components/ProductIcon/index.tsx (hunk 5 above)
Comment (matthprost) [reply packages/icons/src/components/ProductIcon/index.tsx:31]: @chambo-e this has been imported from main project I haven't changed anything, the selector is correct as we use classes for filling the color depending on theme

File: .changeset/light-moose-double.md (hunk 1 above)
Comment (chambo-e) [reply .changeset/light-moose-double.md:2]: Ah yeah nvm, I thought you removed the old Icon :)

**Review by chambo-e — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**